### PR TITLE
Added plotting, updated test set to be last 1,000 instances, separated parquet and pandas loading

### PR DESCRIPTION
**Pull request by emchin**
Hello!

Here's a large push for updated code.

Main things I did:
- Turned pipeline.py into a function pipeline() which I called in main.py 4 times (to create 4 different models)
- Plotted graphs of different models in main.py
- Edited reducer.py to define the test set as the last 1000 values from the parquet dataset <=> keep it consistent across models
- Added docstrings to all methods.

Let me know if you have any questions about what I did! :)

### CODE REVIEW

File: graphs/time.png
Comment (tambat22) [inline graphs/time.png:1]: The runtime bar for `feature_eng_with_parquet` doesn't look correct. I think the graph might be showing the total runtime of the combined tests instead of the individual runtimes.

File: src/risknet/run/main.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+#print(models)
+
+plt.figure(figsize=(20,10)) # 12 inch wide, 4 inch high
+fig, ax = plt.subplots()
+auc_bar = sns.barplot(data=models, x='model', y='auc').set_title('Performance (AUC) for each model')
+ax.bar_label(ax.containers[0], fontsize=10)
+ax.set(xlabel='Models', ylabel='Performance (AUC)')
+ax.set_xticklabels(order, rotation = 20, fontsize=10)
+plt.tight_layout()
+plt.savefig('graphs/aucs.png')
```
Comment (tambat22) [inline src/risknet/run/main.py:74]: I think the `savefig` path should be `../../graphs/aucs.png` instead of `graphs/aucs.png`

**Review by tambat22 — APPROVED** [review]
Looks Good!! These pipeline changes makes it really easy to customize the graphs and models. Thanks!

**Review by rodalbuyeh — CHANGES_REQUESTED** [review]
I like the idea to write docstrings, but there are two issues: 
1. they are in many cases in the wrong position, they should follow after the colon in a definition of a function. 
2. they don't follow any format I am familiar with. I recommend using numpydoc format for docstrings -- https://numpydoc.readthedocs.io/en/latest/format.html -- in this case, in the interest of time, I would recommend providing the functions to a variant of GPT and asking it to generate numpydoc formatted docstrings. For instance, the reduce function docstring should look like this: 

def reduce(fm_root: str, p_true: bool, test_size: int = 1000, split_ratio: List[float] = [0.9, 0.1]) -> pd.DataFrame:
    """
    Concatenates origination data, default labels, and regression labels into a single dataset, 
    then splits this dataset into training, validation, and test sets based on the specified ratios 
    and test set size. The function supports reading from a parquet file for the entire dataset 
    or a subset of the data using pandas for efficiency, based on the `p_true` flag.

    Parameters
    ----------
    fm_root : str
        The file path where the dataset and labels are stored. This path is used as the base 
        to construct full file paths for the origination data, default labels, and regression labels.
    p_true : bool
        Determines whether to use the entire dataset loaded from a parquet file (`True`) 
        or to use the first 10 million rows using pandas (`False`) for efficiency.
    test_size : int, optional
        The number of rows to include in the test set, by default 1000.
    split_ratio : List[float], optional
        The ratio of the train and validation sets, by default [0.9, 0.1]. 
        The list must sum to 1.0 and contain two floats representing the 
        proportion of the dataset to be used for training and validation, respectively.

    Returns
    -------
    pd.DataFrame
        A DataFrame containing the concatenated and split dataset, including training, 
        validation, and test sets, with the test set re-added at the bottom of the DataFrame.

    Notes
    -----
    - The function assumes the presence of specific files at `fm_root`: 'org.parquet', 
      'dev_labels.parquet', and 'dev_reg_labels.parquet' for parquet operations; 
      or their '.pkl' counterparts and a historical data text file for pandas operations.
    - The test set is determined by taking the most recent `test_size` entries based 
      on the 'first_payment_date'.
    - Assumes specific column names for merging and operations, including 'loan_sequence_number',
      'first_payment_date', and various columns to be dropped as defined in `drop_cols`.
    """

**Comment by emchin** [comment]
Thank you for bringing this to my attention! I updated the docstrings accordingly.

**Review by rodalbuyeh — APPROVED** [review]
